src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -0,0 +1,129 @@
+
+  Persistence_interval(dimension_type dim, event_value_type birth, event_value_type death)
+      : dim(dim), birth(birth), death(death) {}
+
+  dimension_type dim;     /**< Dimension of the bar.*/
+  event_value_type birth; /**< Birth index in the current filtration. */
+  event_value_type death; /**< Death index in the current filtration. */
+
+  inline friend std::ostream &operator<<(std::ostream &stream, const Persistence_interval &interval) {
+    stream << "[" << interval.dim << "] ";
+    stream << interval.birth << ", " << interval.death;
+    stream << "\n";
+    return stream;
+  }
+};
+
+}  // namespace persistence_matrix
+}  // namespace Gudhi
+
+namespace std {
+
+template<typename dimension_type, typename event_value_type>
+struct tuple_size<Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type> >
+    : std::integral_constant<std::size_t, 3> {};
+
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr typename tuple_element<I, tuple<event_value_type, event_value_type, dimension_type> >::type& get(
+    Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>& i) noexcept {
+  if constexpr (I == 0) return i.birth;
+  if constexpr (I == 1) return i.death;
+  if constexpr (I == 2) return i.dim;
+  // does not compile if I > 2, because of tuple_element
+}
+
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr const typename tuple_element<I, tuple<event_value_type, event_value_type, dimension_type> >::type& get(
+    const Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>& i) noexcept {
+  if constexpr (I == 0) return i.birth;
+  if constexpr (I == 1) return i.death;
+  if constexpr (I == 2) return i.dim;
+  // does not compile if I > 2, because of tuple_element
+}
+
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr typename tuple_element<I, tuple<event_value_type, event_value_type, dimension_type> >::type&& get(
+    Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>&& i) noexcept {
+  if constexpr (I == 0) return i.birth;
+  if constexpr (I == 1) return i.death;
+  if constexpr (I == 2) return i.dim;
+  // does not compile if I > 2, because of tuple_element
+}
+
+template <size_t I, typename dimension_type, typename event_value_type>
+constexpr const typename tuple_element<I, tuple<event_value_type, event_value_type, dimension_type> >::type&& get(
+    const Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type>&& i) noexcept {
+  if constexpr (I == 0) return i.birth;
+  if constexpr (I == 1) return i.death;
+  if constexpr (I == 2) return i.dim;
+  // does not compile if I > 2, because of tuple_element
+}
+
+template <typename dimension_type, typename event_value_type>
+constexpr dimension_type& get(
```

Comment:
I think we could skip the last 4 overloads, since we can't use the type for birth/death, they look more confusing than helpful.

Reply:
I agree, it is just that if someone had the weird idea to use it for some reasons because it is possible with tuples, it would be there. But I have no problems to remove them.

Reply:
I'd rather they get an error if they try that.